examples/catch2_v2_starter/CMakeLists.txt

```diff
@@ -2,8 +2,12 @@ if (NOT APPROVAL_TESTS_ENABLE_CATCH2_INTEGRATION)
     return()
 endif ()
 
+if(APPROVAL_TESTS_USE_CATCH2V3)
+    return()
+endif()
+
```

Comment:
I tried this, in a version using FetchContent to load the code in this MR. I create a main libvrary which I then link with all my test cases, but found that I had to add 
`target_compile_definitions(approvals_catch_main` PUBLIC APPROVAL_TESTS_USE_CATCH2V3=ON APPROVALS_CATCH)` 
To get it to work

A bit more context

```
FetchContent_Declare(
  Catch2
  GIT_REPOSITORY https://github.com/catchorg/Catch2.git
  GIT_TAG        v3.5.1
  GIT_PROGRESS TRUE
  GIT_SHALLOW TRUE
)

 #
FetchContent_Declare(ApprovalTests
#  GIT_REPOSITORY https://github.com/approvals/ApprovalTests.cpp.git

GIT_REPOSITORY https://github.com/Laguna1989/ApprovalTests.cpp
  GIT_TAG feature/catch3.5
  GIT_PROGRESS TRUE
  GIT_SHALLOW TRUE
)

FetchContent_MakeAvailable(Catch2 ApprovalTests)



add_library(approvals_catch_main catch2_starter_main.cpp)
target_link_libraries(approvals_catch_main
  PUBLIC
  ApprovalTests::ApprovalTests
  Catch2::Catch2WithMain
)
target_compile_definitions(approvals_catch_main PUBLIC APPROVAL_TESTS_USE_CATCH2V3=ON APPROVALS_CATCH)

function(add_catch_test name)
    add_executable(${name} EXCLUDE_FROM_ALL ${ARGN})
    target_link_libraries(${name} PRIVATE  approvals_catch_main)
......
endfunction()
```



And thank you for this, hope it will make its way approvals/master :+1:

Reply:
Hi there @jtpedersen  Thanks for checking this out and giving feedback. 

You assume correctly that it is required to pass `APPROVAL_TESTS_USE_CATCH2V3=ON`. The reason is that this PR supports both catch2 v2 and catch2 v3. 
ApprovalTests.cpp cannot know in advance which version to use. There are some differences (which header to include, different class names, ...) between v2 and v3. Thus it is needed to tell approvaltests via a CMake option which catch version should be used. 

There are multiple options how to pass this

## set flag via cmake command line argument
When you are calling cmake, you cann pass it as a command line argument (this is what is being done on the gitlab action workflow
```
mkdir build
cd build
cmake .. -DAPPROVAL_TESTS_USE_CATCH2V3=ON
cmake --build . --parallel
```

## set flag via cmake command
 This is our suggestion. It can be done on a target level via `target_compile_definitions.

## set option via cmake CPM

An alternative would be to use [CPM](https://github.com/cpm-cmake/CPM.cmake) (CMake Packet Manager). This is considered modern cmake and superior to `FetchContent` for multiple reasons (simplified API, caching, version checking, ...) 

See this minimal cmake example code:
```
message(STATUS "Add CPM.cmake")
# download CPM.cmake
file(
        DOWNLOAD
        https://github.com/cpm-cmake/CPM.cmake/releases/download/v0.38.6/CPM.cmake
        ${CMAKE_BINARY_DIR}/cmake/CPM.cmake
        EXPECTED_HASH SHA256=11c3fa5f1ba14f15d31c2fb63dbc8628ee133d81c8d764caad9a8db9e0bacb07
)
include(${CMAKE_BINARY_DIR}/cmake/CPM.cmake)

message(STATUS "Fetching catch2")
CPMAddPackage(
        NAME catch2
        GITHUB_REPOSITORY catchorg/Catch2
        GIT_TAG v3.5
        OPTIONS
        CATCH_INSTALL_DOCS OFF
        CATCH_INSTALL_HELPERS OFF
        CATCH_BUILD_TESTING OFF
)

message(STATUS "Fetching approvaltests")
CPMAddPackage(
        NAME approvaltests
        GITHUB_REPOSITORY approvals/ApprovalTests.cpp
        GIT_TAG feature/catch3.5.1
        OPTIONS
        APPROVAL_TESTS_USE_CATCH2V3 ON
)
```

Reply:
Thank you for the reply. 

I will look a bit more, however I think the challenge I had was that when including a approvals header, it has an ifdef with respect to catch2 and that will happen when i compile my code, and at that point the configuration of approvals is already done. IE when linking with a approvals the ` APPROVAL_TESTS_USE_CATCH2V3=ON` has to be set.

Reply:
In fact `APPROVAL_TESTS_USE_CATCH2V3=ON` should be set at cmake configure time, which is earlier than link time. If you are using catch2 v2, you do not need to set it at all.